src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
+							// {pathway => {reaction:newIE}}
+							if (parentPathwayWithRLEInstanceEdit != null)
 							{
-								pathway2Reactions2NewIEs.get(parentPathway).add(reaction2newIE);
+								addPathwayToMap(pathwaysMap, currentRLE, newIE, parentPathwayWithRLEInstanceEdit);
 							}
 							else
 							{
-								// If 'Pathway' key doesn't exist, create it and then add to it.
-								Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-								pathway2Reactions2NewIEs.put(parentPathway, reaction2newIESet);
+								addPathwayToMap(pathwaysMap, currentRLE, newIE, parentPathway);
 							}
 						}
 					}
 				}
 			}
+		}
+		return pathwaysMap;
+	}
 
-			printOutput(pathway2Reactions2NewIEs);
+	private static Map<GKInstance, Set<Map<GKInstance, GKInstance>>> processNewRLEs(Set<GKInstance> newRLEs, Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathway2Reactions2NewIEs) throws Exception
+	{
+		
+		Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathwaysMap = pathway2Reactions2NewIEs;
+		
+		// For cases where this the first Release that a RlE has appeared (ie. fetching
+		// for the RlE in the previous DB returned null).
 
-		}
-		catch (SQLException e)
+		// As before, iterate through all RlEs.
+		for (GKInstance newRLE : newRLEs)
 		{
-			e.printStackTrace();
+			// Get non-Reactome IEs from the revised, authored, and reviewed attributes.
+			List<GKInstance> newRLEInstanceEdits = getNonReactomeEventIEs(newRLE);
+
+			// Look through all parents of the RlE, trying to find the new instance edit in
+			// any of their revised, authored, or reviewed attributes.
+			List<GKInstance> parentPathways = (List<GKInstance>) newRLE.getReferers(ReactomeJavaConstants.hasEvent);
+			for (GKInstance parentPathway : parentPathways)
+			{
+				// Get all non-Reactome instance edits in the revised, authored, and reviewed
+				// attributes of the Pathway.
+				List<GKInstance> parentPathwayIEs = getNonReactomeEventIEs(parentPathway);
+
+				boolean hasRLEInstanceEdit = false;
+
+				// Iterate through all new instance edits, trying to find a Pathway containing
+				// them.
+				for (GKInstance rleIE : newRLEInstanceEdits)
+				{
+					for (GKInstance pathwayIE : parentPathwayIEs)
+					{
+						if (rleIE.getDBID().equals(pathwayIE.getDBID()))
+						{
+							// The parent Pathway contained the IE.
+							hasRLEInstanceEdit = true;
+						}
+					}
+					if (hasRLEInstanceEdit)
+					{
+						// If it has the new IE, it might be not be the highest Pathway in the hierarchy
+						// with it.
+						// Continue looking up the hierarchy to try and find the HIGHEST Pathway with
+						// the IE.
+						GKInstance grandparentPathwayWithRLEInstanceEdit = getGrandparentPathwayWithRLEInstanceEdit(parentPathway, rleIE);
+						addPathwayToMap(pathwaysMap, newRLE, rleIE, grandparentPathwayWithRLEInstanceEdit);
+					}
+					else
+					{
+						addPathwayToMap(pathwaysMap, newRLE, rleIE, parentPathway);
+					}
```

Comment:
Same as https://github.com/reactome/doi-suggester/pull/1/files#r610283977.  Why add the `parentPathway` to the map if it doesn't have the RlEInstanceEdit?